### PR TITLE
[HCAP-541] backup mongo db data

### PR DESCRIPTION
**Pull request by vyasworks**

### CODE REVIEW

**Review by arranfw — COMMENTED** [review]
I'd love a walkthrough of the changes made here, it's mostly over my head but what I'm reading makes sense on a surface level

**Review by fwpushan — APPROVED** [review]
👍

**Comment by Zenen-freshworks** [comment]
This PR is going to require some changes now that we have a new instance of mongo running. I'm willing to jump on making the modifications once I'm done with what's on my plate right now.

**Comment by vyasworks** [comment]
okay, I finally was able to resume my work on this yesterday and made some progress. 

The mongo db plugin in `backup-container` maintained by bcgov for OCP 4 has some quirks. 

Issues faced with `backup.sh` : 

- the mongo plugin tries to dump the data as `.sql` not sure if I am missing some config or this is a bug. 
- there seemed to be authentication issues initially but I from Zenen's inputs, adding `--authenticationDatabase` flag fixed it, for our new Mongo containers. 
- exit code 1 without sufficient error logs, makes it really hard to debug, using the terminal 

Solution Implemented. 

- just use `mongodump` directly instead of backup capability `backup.sh`
- backup.sh is still used for creating backup folders

File: openshift/backup-mongo.dc.yml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        requests:
+          storage: '${VERIFICATION_VOLUME_SIZE}'
+  - kind: ConfigMap
+    apiVersion: v1
+    metadata:
+      name: ${NAME}-config
+    data:
+      backup.conf: |
+        mongo=${DB_SERVICE}:27017/${DB_NAME}
+        0 2 * * * default (./backup.sh -s; (mongodump -h hcap-mongo-headless -d $DATABASE_NAME -u $DATABASE_USER -p $DATABASE_PASSWORD --authenticationDatabase $DATABASE_NAME --out /backups/daily/$(date +"%Y-%m-%d") --forceTableScan))
```
Comment (vyasworks) [inline openshift/backup-mongo.dc.yml:37]: This is where all the magic happens.

**Review by fwpushan — APPROVED** [review]
LGTM

**Comment by fwpushan** [comment]
@vyasworks  But I'm seeing some connection failure logs in pod

**Comment by vyasworks** [comment]
> @vyasworks But I'm seeing some connection failure logs in pod

Yes it is expected. 

The `backup.sh` will create folder and fail, followed by the command that does the dump.

**Review by fwkendall — APPROVED** [review]

**Review by Zenen-freshworks — APPROVED** [review]

**Pull request merged**